src/components/select/option/option.test.tsx

```diff
@@ -0,0 +1,209 @@
+});
+
+test("renders with `children` content visible instead of `text` when both are passed", () => {
+  render(
+    <ul>
+      <Option text="foo" value="1">
+        bar
+      </Option>
+    </ul>
+  );
+
+  expect(screen.getByRole("option", { name: "bar" })).toBeVisible();
+});
+
+test("does not render when `hidden` prop is set", () => {
+  render(
+    <ul>
+      <Option text="foo" value="1" hidden />
+    </ul>
+  );
+
+  expect(screen.queryByRole("option", { name: "foo" })).not.toBeInTheDocument();
+});
+
+test("should set the expected `data-` attributes when props passed", () => {
+  render(
+    <ul>
+      <Option text="foo" value="1" data-element="bar" data-role="baz" />
+    </ul>
+  );
+
+  const option = screen.getByRole("option", { name: "foo" });
+
+  expect(option).toHaveAttribute("data-component", "option");
+  expect(option).toHaveAttribute("data-element", "bar");
+  expect(option).toHaveAttribute("data-role", "baz");
+});
+
+describe("when `disabled` prop is set", () => {
+  it("should have expected style", () => {
+    render(
+      <ul>
+        <Option text="foo" value="1" disabled />
+      </ul>
+    );
+    const option = screen.getByRole("option", { name: "foo" });
+
+    expect(option).toHaveStyleRule("color", "var(--colorsUtilityYin030)");
+    expect(option).toHaveStyle("cursor: not-allowed");
+  });
+
+  it("should set the aria-disabled attribute to true", () => {
+    render(
+      <ul>
+        <Option text="foo" value="1" disabled />
+      </ul>
+    );
+    const option = screen.getByRole("option", { name: "foo" });
+
+    expect(option).toHaveAttribute("aria-disabled", "true");
+  });
+
+  it("should not call `onSelect` or `onClick` when the user clicks on the element", async () => {
+    const user = userEvent.setup();
+    const onSelect = jest.fn();
+    const onClick = jest.fn();
+    render(
+      <ul>
+        <Option
+          text="foo"
+          value="1"
+          onClick={onClick}
+          onSelect={onSelect}
+          disabled
+        />
+      </ul>
+    );
+
+    await user.click(screen.getByRole("option", { name: "foo" }));
+
+    expect(onSelect).not.toHaveBeenCalled();
+    expect(onClick).not.toHaveBeenCalled();
+  });
+});
+
+describe("when `disabled` prop is not set", () => {
+  it("should have expected style", () => {
+    render(
+      <ul>
+        <Option text="foo" value="1" />
+      </ul>
+    );
+    const option = screen.getByRole("option", { name: "foo" });
+
+    expect(option).not.toHaveStyleRule("color", "var(--colorsUtilityYin030)");
+    expect(option).toHaveStyle("cursor: pointer");
+  });
+
+  it("should not set the aria-disabled attribute", () => {
+    render(
+      <ul>
+        <Option text="foo" value="1" />
+      </ul>
+    );
+    const option = screen.getByRole("option", { name: "foo" });
+
+    expect(option).not.toHaveAttribute("aria-disabled");
+  });
+
+  it("should call `onSelect` or `onClick` when the user clicks on the element", async () => {
```

Comment:
**suggestion(non-blocking)**: what do you think of this so it reflects calling both event handlers?

```suggestion
  it("should call `onSelect` and `onClick` when the user clicks on the element", async () => {
```

Reply:
I think it works either way here tbh as in the scenario it would call both if it wasn't disabled etc